applications/FSIapplication/custom_utilities/variable_redistribution_utility.cpp

```diff
@@ -59,28 +59,34 @@ void VariableRedistributionUtility::CallSpecializedConvertDistributedValuesToPoi
     const Variable<TValueType>& rDistributedVariable,
     const Variable<TValueType>& rPointVariable)
 {
-    // This function only dispatches the call to the correct specialization
-    Geometry< Node<3> >& rReferenceGeometry = rModelPart.ConditionsBegin()->GetGeometry();
-    const GeometryData::KratosGeometryFamily GeometryFamily = rReferenceGeometry.GetGeometryFamily();
-    const unsigned int PointsNumber = rReferenceGeometry.PointsNumber();
+    // Check if there is any condition in the current partition
+    int n_conds = rModelPart.GetCommunicator().LocalMesh().NumberOfConditions();
+    rModelPart.GetCommunicator().SumAll(n_conds);
 
-    if (GeometryFamily == GeometryData::Kratos_Linear && PointsNumber == 2)
-    {
-        VariableRedistributionUtility::SpecializedConvertDistributedValuesToPoint<GeometryData::Kratos_Linear,2,TValueType>(
-            rModelPart,
-            rDistributedVariable,
-            rPointVariable);
-    }
-    else if (GeometryFamily == GeometryData::Kratos_Triangle && PointsNumber == 3)
-    {
-        VariableRedistributionUtility::SpecializedConvertDistributedValuesToPoint<GeometryData::Kratos_Triangle,3,TValueType>(
-            rModelPart,
-            rDistributedVariable,
-            rPointVariable);
-    }
-    else
-    {
-        KRATOS_ERROR << "Unsupported geometry type with " << PointsNumber << " points." << std::endl;
+    // If there is conditions, this function dispatches the call to the correct specialization
+    if (n_conds != 0){
+        Geometry< Node<3> >& rReferenceGeometry = rModelPart.ConditionsBegin()->GetGeometry();
+        const GeometryData::KratosGeometryFamily GeometryFamily = rReferenceGeometry.GetGeometryFamily();
+        const unsigned int PointsNumber = rReferenceGeometry.PointsNumber();
+
+        if (GeometryFamily == GeometryData::Kratos_Linear && PointsNumber == 2)
```

Comment:
I'm afraid you also need a dummy here, since `SpecializedConvertDistributedValuesToPoint` has a Synchronize inside.

Reply:
Actually, we need to do everything as in the `CallSpecializedDistributePointValues` method. Thanks for noticing!